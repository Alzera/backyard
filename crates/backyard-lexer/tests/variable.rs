use backyard_lexer::lex_eval;

#[test]
fn basic() {
  let tokens = lex_eval("$a").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn this() {
  let tokens = lex_eval("$this").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn bracket() {
  let tokens = lex_eval("${$a}").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}

#[test]
fn multiple() {
  let tokens = lex_eval("$$$a").unwrap();
  insta::assert_yaml_snapshot!(tokens);
}
