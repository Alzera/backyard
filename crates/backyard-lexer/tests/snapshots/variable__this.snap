---
source: crates/backyard-lexer/tests/variable.rs
assertion_line: 12
expression: tokens
snapshot_kind: text
---
- token_type: This
  value: this
  line: 1
  column: 0
  offset: 0
